pandas/core/indexes/datetimes.py

```diff
@@ -1968,14 +1974,41 @@ def tz_localize(self, tz, ambiguous='raise', errors='raise'):
             .. deprecated:: 0.15.0
                Attempt to infer fall dst-transition hours based on order
 
+
         Returns
         -------
-        localized : DatetimeIndex
+        DatetimeIndex
+            It returns naive time or aware time
+
+        Examples
+        --------
+        >>> tz_naive = pd.date_range('2018-03-01', '2018-03-03')
+
+        >>> tz_naive
+        DatetimeIndex(['2018-03-01', '2018-03-02', '2018-03-03'],
+        dtype='datetime64[ns]', freq='D')
+
+        Localize DatetimeIndex in US/Eastern time zone.
+
+        >>> tz_aware = tz_naive.tz_localize(tz='US/Eastern')
+
+        >>> tz_aware
+        DatetimeIndex(['2018-03-01 00:00:00-05:00',
+        '2018-03-02 00:00:00-05:00', '2018-03-03 00:00:00-05:00'],
+        dtype='datetime64[ns, US/Eastern]', freq='D')
+
+        Localize aware time zone into naive time zone.
+
+        >>> tz_aware.tz_localize(None)
+        DatetimeIndex(['2018-03-01', '2018-03-02', '2018-03-03'],
+        dtype='datetime64[ns]', freq='D')
+
 
         Raises
         ------
         TypeError
             If the DatetimeIndex is tz-aware and tz is not None.
+
```

Comment:
I think we don't leave this line here, the validation script probably complains.